ciao-launcher/main.go

```diff
@@ -318,6 +315,16 @@ func processCommand(conn serverConn, cmd *cmdWrapper, ovsCh chan<- interface{})
 	}
 }
 
+func startNetwork() error {
+	if networking.Enabled() {
```

Comment:
This code is not quite right.  The original networking code also selected on the signal channel so that the code that inits the network could not block a launcher exit via CTRL-C or SIGTERM.  This code has been deleted.  Launcher will still be able to shutdown but it will no longer be able to shutdown cleanly.

You don't have the signal channel in the connectToServer go routine but you do have doneCh which is closed when we received a signal

```
func startNetwork(doneCh chan struct{}) error {
	  if networking.Enabled() {
                ctx, _ := context.WithCancel(context.Background())
		ch := initNetworking(ctx)
		select {
		case <-doneCh:
			glog.Info("Received terminating signal.  Quitting")
			cancelFunc()
			return fmt.Error("Shutting down")
		case err := <-ch:
			if err != nil {
				glog.Errorf("Failed to init network: %v\n", err)
				return err
			}
		}
}
```